core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -64,6 +65,7 @@ public class HTTPClient implements RESTClient {
   private final CloseableHttpClient httpClient;
   private final ObjectMapper mapper;
   private final Map<String, String> baseHeaders;
+  private Map<String, String> responseHeaders;
```

Comment:
I don't think this is going to work since this is at the client-level and response headers are at the request-level.  We should probably follow the consumer pattern (e.g. `Consumer<Map<String, String>> responseHeaders`)

Reply:
yes this isn't an ideal approach, since it will only return whatever the response headers were from the **last** request that was executed

Reply:
@danielcweeks I've updated the PR with your proposal